diff --git a/datanommer.models/datanommer/models/__init__.py b/datanommer.models/datanommer/models/__init__.py
index c7b4561..902a01e 100644
--- a/datanommer.models/datanommer/models/__init__.py
+++ b/datanommer.models/datanommer/models/__init__.py
@@ -181,7 +181,12 @@ class JSONEncodedDict(TypeDecorator):
 
     def process_result_value(self, value, dialect):
         if value is not None:
-            value = json.loads(value)
+            try:
+                value = json.loads(value)
+            except TypeError as e:
+                log.exception(e)
+                log.error(repr(value))
+                raise
         return value
 
     def coerce_compared_value(self, op, value):
